src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1515,26 +1515,62 @@ export class SavedObjectsRepository {
   }
 
   /**
-   * Increases a counter field by one. Creates the document if one doesn't exist for the given id.
+   * Increments all the specified counter fields by one. Creates the document
+   * if one doesn't exist for the given id.
+   *
+   * @remarks
+   * When using incrementCounter for collecting usage data, you need to ensure
+   * that usage collection happens on a best-effort basis and doesn't
+   * negatively affect your plugin or users (see the example):
+   *  - Swallow any exceptions thrown from the incrementCounter method and log
+   *    a message in development.
+   *  - Don't block your application on the incrementCounter method (e.g.
+   *    don't use `await`)
+   *
+   * When supplying a field name like `stats.api.counter` the field name will
+   * be used as-is to create a document like:
+   *   `{attributes: {'stats.api.counter': 1}}`
+   * It will not create a nested structure like:
+   *   `{attributes: {stats: {api: {counter: 1}}}}`
```

Comment:
NIT: Looking at `docs/development/core/server/kibana-plugin-core-server.savedobjectsrepository.incrementcounter.md`, it doesn't include this second remark.

Reply:
good catch!